libraries/AP_HAL_ChibiOS/hwdef/CarbonixCommon/scripts/cx_built_in_test.lua

```diff
@@ -59,8 +59,45 @@ assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 1), 'could not add '
 --]]
 local PREARM_BYPASS = bind_add_param('PREARM_DIS', 1, 0)
 
+-- fetches the aircraft configuration from the config file, if it exists
+local function get_aircraft_config(file_name)
+    local model = nil
+    local model_version = nil
+
+    -- Open file try and read header
+    local file = io.open(file_name,"r")
+    if not file then
+        cx_msg:send(cx_msg.MAV_SEVERITY.INFO, "config file not found")
+        return
+    end
+    while true do
+        local line = file:read()
+        if not line then
+            -- empty file or reached end of file
+            cx_msg:send(cx_msg.MAV_SEVERITY.INFO, "missing model info in cfg")
+            break
+        end
+        
+        -- Extract model using pattern matching
+        if string.match(line, "<model>(.-)</model>") then
+            model = string.match(line, "<model>(.-)</model>")
+        end
+        -- Extract model_version using pattern matching
+        if string.match(line, "<model_version>(.-)</model_version>") then
+            model_version = string.match(line, "<model_version>(.-)</model_version>")
+        end
+
+        if model and model_version then
+            cx_msg:send(cx_msg.MAV_SEVERITY.INFO, "config (" .. model .. "_" .. model_version .. ".xml)")
+            return
+        end
```

Comment:
Should have an else and print a message like you did above for if `line` is nil

Reply:
Avoided this print, because we are parsing an xml file.
But this is handled by this check at the top : 
```
        if not line then
            -- empty file or reached end of file
            cx_msg:send(cx_msg.MAV_SEVERITY.INFO, "missing model info in cfg")
            break
        end
```